ow_sim_tests/scripts/test_arm_stop.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python
+
+# The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+# Research and Simulation can be found in README.md in the root directory of
+# this repository.
+
+import rospy
+import roslib
+import unittest
+import owl_msgs.msg
+
+from action_testing import *
+
+PKG = 'ow_sim_tests'
+TEST_NAME = 'arm_stop'
+roslib.load_manifest(PKG)
+
+class TestArmStop(unittest.TestCase):
+
+  @classmethod
+  def setUpClass(cls):
+    rospy.init_node("arm_stop_test")
+
+    set_ignore_action_checks(True)
+    # proceed with test only when ros clock has been initialized
+    while rospy.get_time() == 0:
+      rospy.sleep(0.1)
+    
+  def test_01_arm_stop(self):
+
+    client_stop = actionlib.SimpleActionClient('ArmStop', owl_msgs.msg.ArmStopAction)
+
+    connected_stop = client_stop.wait_for_server(timeout = rospy.Duration(50))
+
+    client_stop.send_goal(owl_msgs.msg.ArmStopGoal())
+    client_stop.wait_for_result()
+
+    result = client_stop.get_result()
+
+    self.assertTrue(result, "No arm trajectory to stop")
+
```

Comment:
[optional]
If you'd like, the **action_testing.py** can be used here, but only here since the other units are calling actions asynchronously. But here, you call a single action synchronously. 

```suggestion
    arm_stop_result = test_arm_action(self,
      'ArmStop', owl_msgs.msg.ArmStopAction,
      owl_msgs.msg.ArmStopGoal(),
      TEST_NAME, "test_01_arm_stop",
      server_timeout = 50.0 # (seconds) first action call needs longer timeout,
      expected_end_state = actionlib.GoalStatus.SUCCEEDED
    )

```

There is a drawback, which is `test_arm_action` returns the action result, which I previously thought contained the status and status message of the action, but in fact does not. You'll see in **action_testing.py** on line 133, there is a TODO to instead return the client object. 

So if you would rather test the result message, the change could also just be what I discuss in the next comment down. 

```suggestion
    client_stop = actionlib.SimpleActionClient('ArmStop', owl_msgs.msg.ArmStopAction)

    connected_stop = client_stop.wait_for_server(timeout = rospy.Duration(50))

    client_stop.send_goal(owl_msgs.msg.ArmStopGoal())
    client_stop.wait_for_result()

    result = client_stop.get_result()

    self.assertEquals(client_stop.get_goal_status_text(), "No arm trajectory to stop")

```

Reply:
Thank you so much for the correction. I also make changes on ` expected_end_state = actionlib.GoalStatus.SUCCEEDED`, in my understanding should be `expected_end_state = actionlib.GoalStatus.ABORTED` since `test_01` does not stop any action, therefore we expect to see the end state as `ABORTED`.

Reply:
Yes! Good catch, I mean for that to be ABORTED. Fixed in the original message.